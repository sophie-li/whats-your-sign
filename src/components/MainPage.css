@font-face {
  font-display: swap;
  font-family: 'BebasNeue-Regular';
  font-style: normal;
  font-weight: normal;
  src: local('BebasNeue-Regular', url('BebasNeue-Regular.woff') format('woff'));
}

h1 {
  font-family: 'BebasNeue-Regular';
  font-weight: 200;
  color: #ffffff;
}

img {
  height: 100px;
  width: 100px;
}

.magic-img {
  height: 50px;
  width: 50px;
  padding: 10px;
}

.magic-img:hover {
  cursor: pointer;
}

.square-color {
  display: inline-block;
  height: 13px;
  width: 13px;
  margin: 3px 0 0 5px;
  border: 1px solid black;
  clear: both;
}

/* style mobile first */
@media only screen and (max-width: 640px) {
  .card {
    padding: 1rem;
    margin: 5px 10px;
  }
}

/* tablet */
@media only screen and (min-width: 641px) and (max-width: 1024px) {
  .card {
    width: 45%;
    padding: 1rem;
    margin-right: 0.5rem;
  }
}

/* desktop */
@media only screen and (min-width: 1025px) {
  .card {
    width: 25%;
    padding: 1rem;
    margin-right: 0.5rem;
  }
}
