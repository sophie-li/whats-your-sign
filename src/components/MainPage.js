import React, { Component } from 'react';

import CardList from '../components/CardList';
import ErrorBoundry from '../components/ErrorBoundry';
import Header from '../components/Header';
import Scroll from '../components/Scroll';
import SearchBox from '../components/SearchBox';

import './MainPage.css';

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestSigns();
  }

  filterSigns = () => {
    return this.props.signs.filter((sign) => {
      return sign.name
        .toLowerCase()
        .includes(this.props.searchField.toLowerCase());
    });
  };

  render() {
    const { signs, onSearchChange, isPending } = this.props;

    return (
      <div className='tc'>
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          {isPending ? (
            <h1>Loading</h1>
          ) : (
            <ErrorBoundry>
              <CardList signs={this.filterSigns(signs)} />
            </ErrorBoundry>
          )}
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
