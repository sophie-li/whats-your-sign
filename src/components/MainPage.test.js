import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestSigns: jest.fn(),
    signs: [],
    searchField: '',
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it('expect to render MainPage', () => {
  expect(wrapper).toMatchSnapshot();
});

it('filters signs correctly, giving empty results', () => {
  const mockProps2 = {
    onRequestSigns: jest.fn(),
    signs: [{ id: 2, name: 'aries' }],
    searchField: 'b',
    isPending: false,
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);

  expect(wrapper2.instance().filterSigns()).toEqual([]);
});

it('filters signs correctly, giving matching results', () => {
  const mockProps3 = {
    onRequestSigns: jest.fn(),
    signs: [{ id: 2, name: 'aries' }],
    searchField: 'a',
    isPending: false,
  };
  const wrapper3 = shallow(<MainPage {...mockProps3} />);

  expect(wrapper3.instance().filterSigns()).toEqual([{ id: 2, name: 'aries' }]);
});
