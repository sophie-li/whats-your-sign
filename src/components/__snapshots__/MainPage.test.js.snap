// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        signs={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
